src/ui/public/vislib_vis_type/vislib_renderbot.js

```diff
@@ -46,11 +47,8 @@ module.exports = function VislibRenderbotFactory(Private) {
   VislibRenderbot.prototype.buildChartData = buildChartData;
   VislibRenderbot.prototype.render = function (esResponse) {
     this.chartData = this.buildChartData(esResponse);
-    // to allow legend to render first (wait for angular digest cycle to complete)
-    return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve(this.vislibVis.render(this.chartData, this.uiState));
-      });
+    return AngularPromise.delay(1).then(() => {
```

Comment:
Is the goal here still to make sure `render` gets called after the current digest cycle, so that the legend has already been rendered? If so, is this solution guaranteed to work or is there the possibility of a race condition?